### DISCUSSION TITLE
Installing and running K3OS off a SLC-based USB flash drive?

### DISCUSSION THREAD

**Discussion by victorhooi** [k3s-io/k3s#3641] [General]
I'm hoping to setup a small homelab, with 3 nodes.

The plan is to install K3OS on a bunch of NUCs, and then use Ceph (or maybe Longhorn?) as the storage backend:

https://www.reddit.com/r/kubernetes/comments/oj3xup/k3s_homelab_with_3_nodes_and_1_physical_disk_what/

However, each NUC only has a single physical drive (M.2)

My question is - would it be possible to install K3OS onto a USB disk, and use that as the boot drive - then dedicate the inbuilt M.2 to Ceph/Longhorn?

(This would be an industrial grade SLC-based USB Flash Drive - which should give it the added longevity/reliability needed to work as a boot drive).

If so, how much space would you need on the USB stick?

**Comment by dweomer** [comment]
@victorhooi apologies for the late-ish response to this but I run k3OS in my homelab with rancher and longhorn. I currently have 16GB Optane nvme as the root disk and mount the ssd via `boot_cmd` to `/var/lib/longhorn` because I also develop against this cluster at times for https://github.com/rancher/kim I do occasionally fill up the disk. I have purchased some 32GB Optanes and hope to install those soon for some more space.

With a super-fast USB rive your IO throughput should be fast enough but I wouldn't try anything smaller than 32GB. The amount of space that you need is dependent on the content that drives the workloads that you intend to run. Thinking Rancher + some ML? Consider much fatter drives.
